unblob/processing.py

```diff
@@ -560,13 +560,42 @@ def _process_chunks(
                 self.result.add_report(chunk.as_report(extraction_reports=[]))
             return
 
+        is_whole_file_chunk = len(outer_chunks) + len(unknown_chunks) == 1
+        if is_whole_file_chunk:
+            # skip carving, extract directly the whole file (chunk)
+            carved_path = self.task.path
+            for chunk in outer_chunks:
+                self._extract_chunk(
+                    carved_path,
+                    chunk,
+                    self.config.get_extract_dir_for(carved_path),
+                    # since we do not carve, we want to keep the input around
+                    remove_extracted_input=False,
```

Comment:
That is true only if the user set the `-k` option or the `--keep-extracted-chunks`. No ?

Reply:
It is actually confusing because in `_extract_chunk` you do:

```py
if remove_extracted_input and not self.config.keep_extracted_chunks:
```

Reply:
> That is true only if the user set the `-k` option or the `--keep-extracted-chunks`. No ?

No, we keep the carved chunks in this case, as here the file is from the extraction (*whole file chunk*), which we do not carve, thus we want to keep input file even if extracted (we want to remove only carved files).

> It is actually confusing because in `_extract_chunk` you do:
> 
> ```python
> if remove_extracted_input and not self.config.keep_extracted_chunks:
> ```

Thanks, this is good catch, moved the condition to the other call site in `_carve_then_extract_chunk`, where it belongs.